pkg/collector/listeners/kubelet.go

```diff
@@ -0,0 +1,217 @@
+				}
+				for _, containerID := range expiredContainerList {
+					l.removeService(ID(containerID))
+				}
+			}
+		}
+	}()
+}
+
+func (l *KubeletListener) Stop() {
+	l.ticker.Stop()
+	l.stop <- true
+}
+
+func (l *KubeletListener) processNewPod(pod *kubelet.Pod) {
+	for _, container := range pod.Status.Containers {
+		l.createService(ID(container.ID), pod)
+	}
+}
+
+func (l *KubeletListener) createService(id ID, pod *kubelet.Pod) {
+	svc := PodContainerService{
+		ID: id,
+	}
+	podName := pod.Metadata.Name
+
+	// AD Identifiers
+	var containerName string
+	for _, container := range pod.Status.Containers {
+		if container.ID == string(svc.ID) {
+			svc.ADIdentifiers = append(svc.ADIdentifiers, container.Name)
+			long, short, _, err := docker.SplitImageName(container.Image)
+			if err != nil {
+				log.Warnf("Error while spliting image name: %s", err)
+			}
+			if len(long) > 0 {
+				svc.ADIdentifiers = append(svc.ADIdentifiers, long)
+			}
+			if len(short) > 0 && short != long && short != container.Name {
+				svc.ADIdentifiers = append(svc.ADIdentifiers, short)
+			}
+			containerName = container.Name
+			break
+		}
+	}
+
+	// Hosts
+	podIp := pod.Status.PodIP
+	if podIp == "" {
+		log.Errorf("Unable to get pod %s IP", podName)
+	}
+	svc.Hosts = map[string]string{"pod": podIp}
+
+	// Ports
+	var ports []int
+	for _, container := range pod.Spec.Containers {
+		if container.Name == containerName {
+			for _, port := range container.Ports {
+				ports = append(ports, port.ContainerPort)
+			}
+			break
+		}
+	}
+	svc.Ports = ports
+	if len(svc.Ports) == 0 {
+		// Port might not be specified in pod spec
+		log.Errorf("Failed to get ports for pod %s", podName)
+	}
+
+	// Tags
+	_, err := svc.GetTags()
+	if err != nil {
+		log.Errorf("Failed to get tags for pod %s: %s", podName, err)
+	}
+
+	l.m.Lock()
+	l.services[ID(id)] = &svc
+	l.m.Unlock()
+
+	l.newService <- &svc
+}
+
+func (l *KubeletListener) removeService(cID ID) {
+	l.m.RLock()
+	svc, ok := l.services[cID]
+	l.m.RUnlock()
+
+	if ok {
+		l.m.Lock()
+		delete(l.services, cID)
+		l.m.Unlock()
+
+		l.delService <- svc
+	} else {
+		log.Debugf("Container %s not found, not removing", cID)
+	}
+}
+
+// GetID returns the service ID
+func (s *PodContainerService) GetID() ID {
+	return s.ID
+}
+
+// GetADIdentifiers returns the service AD identifiers
+func (s *PodContainerService) GetADIdentifiers() ([]string, error) {
+	return s.ADIdentifiers, nil
+}
+
+// GetHosts returns the pod hosts
+func (s *PodContainerService) GetHosts() (map[string]string, error) {
+	return s.Hosts, nil
+}
+
+// GetPid inspect the container an return its pid
+func (s *PodContainerService) GetPid() (int, error) {
+	return -1, ErrNotSupported
+}
+
+// GetPorts returns the container's ports
+func (s *PodContainerService) GetPorts() ([]int, error) {
+	return s.Ports, nil
+}
+
+// GetTags retrieves tags using the Tagger
+func (s *PodContainerService) GetTags() ([]string, error) {
+	tags, err := tagger.Tag(string(s.ID), true)
```

Comment:
we use low cardinality for AD, so `Tag` should be called with `false`

Additionnaly, as Tag already returns a valid empty list on error, you can just write `return tagger.Tag(string(s.ID), false)`